src/lib/pickles/verify.ml

```diff
@@ -49,210 +40,70 @@ let verify_heterogenous (ts : Instance.t list) =
     ((fun (lab, b) -> if not b then r := lab :: !r), result)
   in
   [%log internal] "Compute_plonks_and_chals" ;
-  let in_circuit_plonks, computed_bp_chals =
+  let computed_bp_chals, deferred_values =
     List.map ts
       ~f:(fun
            (T
              ( _max_proofs_verified
              , _statement
              , key
-             , app_state
+             , _app_state
              , T
-                 { statement
-                   (* TODO
-                      ; prev_x_hat = (x_hat1, _) as prev_x_hat
-                   *)
+                 { statement =
+                     { proof_state
+                     ; messages_for_next_step_proof =
+                         { old_bulletproof_challenges; _ }
+                     }
+                     (* TODO
+                        ; prev_x_hat = (x_hat1, _) as prev_x_hat
+                     *)
```

Comment:
Since we kept the TODO here, could we elaborate on it?

Reply:
It should be deleted

Reply:
It should be deleted